sklearn/feature_selection/_sequential.py

```diff
@@ -29,11 +31,29 @@ class SequentialFeatureSelector(SelectorMixin, MetaEstimatorMixin, BaseEstimator
     estimator : estimator instance
         An unfitted estimator.
 
-    n_features_to_select : int or float, default=None
-        The number of features to select. If `None`, half of the features are
-        selected. If integer, the parameter is the absolute number of features
-        to select. If float between 0 and 1, it is the fraction of features to
-        select.
+    n_features_to_select : int or float, default='warn'
+        If 'auto', the behaviour depends on the `tol` parameter:
+
+        - if `tol` is not None, then features are selected until the score
+          improvement does not exceed `tol`.
+        - otherwise, half of the features are selected.
+
+        If integer, the parameter is the absolute number of features to select.
+        If float between 0 and 1, it is the fraction of features to select.
+
+        .. versionadded:: 1.0
+            The default changed from None to 'warn' in 1.0 and will become
+            'auto' in 1.2. To keep the same behaviour as with `None`, you
+            should manually set `n_features_to_select='auto'` and
+            set `tol=None`
+
+    tol : float, default=0.0
+        If the score is not incremented by at least `tol` between two
+        consecutive feature addition or removal, stop adding or removing even
+        if `n_features_to_select` has not been reached. `tol` is enabled only
+        when `n_features_to_select` is `'auto'`.
```

Comment:
```suggestion
        when `n_features_to_select` is `"auto"`.
```

Reply:
I assume that we should as well document the `None` option. From the deprecation warning, I was under the impression that it allows for backward compatibility.

Reply:
add docstring about None option with https://github.com/scikit-learn/scikit-learn/pull/20145/commits/744c2358e0ab4b1a79fbfd9e55cc134ab7187cce